npm-packages.md

```diff
@@ -0,0 +1,46 @@
+# Publishing and using `@guardian` NPM packages
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Using
+
+-   Make sure you include any `@guardian` packages for transpilation in your build process
+    -   e.g. your [`webpack.config.js`](https://github.com/webpack/webpack/issues/2031#issuecomment-219040479) might including something that looks like this:
+
+```js
+module: {
+	rules: [
+		{
+			test: /\.m?(j|t)sx?$/,
+			use: {
+				loader: 'babel-loader',
+			},
+			exclude: {
+				test: /node_modules/,
+
+				// don't exclude '@guardian' node_modules
+				exclude: /@guardian\//,
+			},
+		},
+	]
+}
+```
+
+## Publishing
+
+### Use `@guardian`
+
+-   Publish to NPM using the [`@guardian`](https://www.npmjs.com/org/guardian) scope
```

Comment:
Have we decided that `npm` is the place to standardise, and we should migrate away from any attempts to use GitHub URLs or GitHub Packages?

If so, do we have any more specific recommendations for _how_ to publish to `npm`? E.g. using `np`, publishing from `main` etc.?

Reply:
> Have we decided that npm is the place to standardise, and we should migrate away from any attempts to use GitHub URLs or GitHub Packages?

I'm not convinced on the value vs confusion this would cause? I'd rather the consistency and the effort to switch *all* to github or github packages - does it seem worth it?

Reply:
> If so, do we have any more specific recommendations for how to publish to npm? E.g. using np, publishing from main etc.?

Oh, I just noticed that in moving to a separate file we lost a recommendation about using `np` to publish. @sndrs I'm not sure if that was intentional? We used to have this:

```
Use [np](https://www.npmjs.com/package/np) to publish packages. Add it as a project dependency and call it via a `release` script in your `package.json`.
```

Reply:
> I'd rather the consistency and the effort to switch _all_ to github or github packages - does it seem worth it?

On *GitHub_ dependencies*, this means using the [`github:` prefix and `#<commit-ish>`](https://docs.npmjs.com/cli/v6/commands/npm-install) to specify a repo and branch/commit/tag. This comes with advantages and disadvantages. We're using it [on AR](https://github.com/guardian/apps-rendering/blob/bd7214635ea43813673e2fb40db783e45069bd58/package.json#L45) for a few things - other projects like DCR would have to decide if that would work for them as well.

On *GitHub packages*, perhaps we should do another spike to see if they're easier to use since @oliverlloyd last looked into them? AFAIK there are some nice upsides (GitHub auth so no need for a separate npm account, ability to publish via GitHub Actions, etc.), but some awkward downsides (auth required to install packages?).

Reply:
> Oh, I just noticed that in moving to a separate file we lost a recommendation about using np to publish. @sndrs I'm not sure if that was intentional?

no an accident, will restore it